opencsp/app/sofast/test/test_ProcessSofastFixed.py

```diff
@@ -28,26 +30,78 @@ class TestProcessSofastFixed(unittest.TestCase):
-        file_ori = join(dir_sofast_common, 'spatial_orientation.h5')
-        file_dot_locs = join(dir_sofast_fixed, 'data_measurement/fixed_pattern_dot_locations.h5')
-        file_meas = join(dir_sofast_fixed, 'data_measurement/measurement_facet.h5')
-        file_exp = join(dir_sofast_fixed, 'data_expected/calculation_facet.h5')
-
-        cls.save_dir = join(opencsp_code_dir(), 'app/sofast/test/data/output/process_sofast_fixed_single_facet')
+        cls.dir_sofast_fixed = join(opencsp_code_dir(), 'test/data/sofast_fixed')
+        cls.dir_sofast_common = join(opencsp_code_dir(), 'test/data/sofast_common')
+        cls.save_dir = join(opencsp_code_dir(), 'app/sofast/test/data/output/process_sofast_fixed')
         ft.create_directories_if_necessary(cls.save_dir)
 
         lt.logger(join(cls.save_dir, 'sofast_fixed_process_log.txt'), lt.log.ERROR)
 
-        surface = Surface2DParabolic(initial_focal_lengths_xy=(150.0, 150), robust_least_squares=False, downsample=1)
+        cls.sofast_single_facet: ProcessSofastFixed = None
+        cls.sofast_facet_ensemble: ProcessSofastFixed = None
+        cls.exp_slopes_xy_single_facet: np.ndarray = None
+        cls.exp_slopes_xy_facet_ensemble: list[np.ndarray] = None
+        cls.exp_facet_pointing_trans: list[np.ndarray] = None
+
+        cls._process_facet_ensemble(cls)
+        cls._process_single_facet(cls)
+
+    def _process_facet_ensemble(self):
+        # Definitions
+        file_camera = join(self.dir_sofast_common, "camera_sofast.h5")
+        file_facet = join(self.dir_sofast_common, "Facet_split_NSTTF.json")
+        file_ensemble = join(self.dir_sofast_common, 'Ensemble_split_NSTTF_facet.json')
+        file_ori = join(self.dir_sofast_common, 'spatial_orientation.h5')
+        file_dot_locs = join(self.dir_sofast_fixed, 'data_measurement/fixed_pattern_dot_locations.h5')
+        file_meas = join(self.dir_sofast_fixed, 'data_measurement/measurement_facet_ensemble.h5')
+        file_exp = join(self.dir_sofast_fixed, 'data_expected/calculation_facet_ensemble.h5')
 
         # Load data
         camera = Camera.load_from_hdf(file_camera)
+        ensemble_data = DefinitionEnsemble.load_from_json(file_ensemble)
+        facet_data = [DefinitionFacet.load_from_json(file_facet)] * ensemble_data.num_facets
+        dot_locs = DotLocationsFixedPattern.load_from_hdf(file_dot_locs)
+        orientation = SpatialOrientation.load_from_hdf(file_ori)
+        measurement = MeasurementSofastFixed.load_from_hdf(file_meas)
+        surfaces = [
+            Surface2DParabolic(initial_focal_lengths_xy=(150.0, 150), robust_least_squares=False, downsample=1)
+        ] * ensemble_data.num_facets
+
+        # Load expected data
+        self.exp_slopes_xy_facet_ensemble = []
+        self.exp_facet_pointing_trans = []
+        for idx_facet in range(ensemble_data.num_facets):
+            datasets = [
+                f'DataSofastCalculation/facet/facet_{idx_facet:03d}/SlopeSolverData/slopes_facet_xy',
```

Comment:
re: 03d, `idx_facet` cannot exceed 999? if it can exceed, `DefinitionEnsemble` constructor should check for upper bound and raise an exception if exceeded.

Reply:
Same comment as before. It is very unlikely to be greater than 999. I'd advocate for leaving it. But I'm definitely open to other approaches that I haven't thought of yet.